src/components/Common/SearchInput.tsx

```diff
@@ -183,43 +236,22 @@ const SearchByMultipleFields: React.FC<SearchByMultipleFieldsProps> = ({
     }
   }, [searchValue]);
 
-  const renderSearchInput = useMemo(() => {
-    switch (selectedOption.type) {
-      case "phone":
-        return (
-          <div className="relative">
-            <PhoneInput
-              id={id}
-              name={selectedOption.key}
-              placeholder={selectedOption.placeholder}
-              value={searchValue}
-              onChange={(value) => setSearchValue(value ?? "")}
-              className={inputClassName}
-              autoFocus={autoFocus}
-            />
-            {!isSingleOption && <KeyboardShortcutHint open={open} />}
-          </div>
-        );
-      default:
-        return (
-          <div className="relative">
-            <Input
-              id={id}
-              type="text"
-              placeholder={selectedOption.placeholder}
-              ref={inputRef}
-              value={searchValue}
-              onChange={(event) => setSearchValue(event.target.value)}
-              className={cn(
-                "grow border-none shadow-none focus-visible:ring-0",
-                inputClassName,
-              )}
-            />
-            {!isSingleOption && <KeyboardShortcutHint open={open} />}
-          </div>
-        );
-    }
-  }, [selectedOption, searchValue, t, inputClassName, open]);
+  const renderInput = useMemo(
+    () => (
+      <SearchInputFieldRenderer
+        selectedOption={selectedOption}
+        searchValue={searchValue}
+        setSearchValue={setSearchValue}
+        inputRef={inputRef}
+        inputClassName={inputClassName}
+        autoFocus={autoFocus}
+        isSingleOption={isSingleOption}
+        open={open}
+        {...props}
+      />
+    ),
+    [selectedOption, searchValue, inputClassName, open],
+  );
```

Comment:
react will memoize this automatically! why do we need to use `useMemo` for components? 🤔